bool platform
bool arm
---
bool finished